src/v/cluster/node_status_backend.cc

```diff
@@ -93,7 +93,9 @@ ss::future<> node_status_backend::drain_notifications_queue() {
 
 ss::future<> node_status_backend::handle_members_updated_notification(
   model::node_id node_id, model::membership_state state) {
-    if (state == model::membership_state::active) {
+    switch (state) {
+    case model::membership_state::active:
```

Comment:
this looks fine but I thought you mentioned something about issuing notifications differently when loading from a snapshot (thats probably going to be a little more complicated than this patch) but just wanted to check..

Reply:
So I thought for a while and decided to keep the current `members_table` behavior. This is essentially a difference between edge-triggered and level-triggered events and I think it is better to stick to one or another (level-triggered in this case). Added a clarifying comment to `members_table`.

Reply:
right, that makes sense.